orc8r/cloud/go/services/metricsd/prometheus/prometheus-cache/cache/cache.go

```diff
@@ -147,18 +148,51 @@ func (c *MetricCache) clearMetrics() {
 	c.metricFamiliesByName = make(map[string]*familyAndMetrics)
 }
 
-func (c *MetricCache) exposeMetrics(metricFamiliesByName map[string]*familyAndMetrics) string {
-	respStr := strings.Builder{}
+func (c *MetricCache) exposeMetrics(metricFamiliesByName map[string]*familyAndMetrics, workers int) string {
+	fams := make(chan *familyAndMetrics, workers)
+	results := make(chan string, workers)
+	respStr := make(chan string, 1)
+
+	waitGroup := &sync.WaitGroup{}
+
+	for i := 0; i < workers; i++ {
+		waitGroup.Add(1)
+		go processFamilyWorker(fams, results, waitGroup)
+	}
+
+	go processFamilyStringsWorker(results, respStr)
+
 	for _, fam := range metricFamiliesByName {
+		fams <- fam
+	}
+
+	close(fams)
+	waitGroup.Wait()
```

Comment:
can you add a timeout here to cover cases like workers panicking and not marking themselves as done

Reply:
The defer statement always executes no matter how the method ends right?

Reply:
> Panic is a built-in function that stops the ordinary flow of control and begins panicking. When the function F calls panic, execution of F stops, any deferred functions in F are executed normally, and then F returns to its caller.

https://blog.golang.org/defer-panic-and-recover